internal/lsp/signature.go

```diff
@@ -0,0 +1,33 @@
+package lsp
+
+import (
+	"github.com/hashicorp/hcl-lang/lang"
+	"github.com/hashicorp/terraform-ls/internal/mdplain"
+	lsp "github.com/hashicorp/terraform-ls/internal/protocol"
+)
+
+func ToSignatureHelp(signature *lang.FunctionSignature) *lsp.SignatureHelp {
+	if signature == nil {
+		return nil
+	}
+
+	parameters := make([]lsp.ParameterInformation, 0)
```

Comment:
```suggestion
	parameters := make([]lsp.ParameterInformation, 0, len(signature.Parameters))
```
Nitpick: We can allocate the exact capacity since we know it upfront.